GoRogue/MapGeneration/Region.cs

```diff
@@ -9,17 +10,56 @@
 
 namespace GoRogue.MapGeneration
 {
+    /// <summary>
+    /// Event fired when a region's Area is changed.
+    /// </summary>
+    [PublicAPI]
+    public class RegionAreaChanged : EventArgs
+    {
+        /// <summary>
+        /// The region whose area is changing
+        /// </summary>
+        public readonly Region Item;
```

Comment:
Just to clarify, this field should be generally unnecessary because its value will always be the same as the value in the `sender` parameter given to the event handler.  Some events in the library include this more strongly typed field anyway, some do not.  It's whichever you prefer; but outside of the difference in typing it is generally redundant.